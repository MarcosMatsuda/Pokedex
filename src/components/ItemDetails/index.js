// import React from 'react';
// import { SafeAreaView, Text, FlatList, View } from 'react-native';
// import { useSelector } from 'react-redux';
// import { useNavigation } from '@react-navigation/native';

// import { ProgressiveDiscount } from '~/components';

// import { Container } from './styles';

// const ItemDetails = ({ title }) => (
//   <View>
//     <Text>{title}</Text>
//   </View>
// );

// export { ItemDetails };
