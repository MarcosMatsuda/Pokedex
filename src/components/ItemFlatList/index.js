import React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { useNavigation } from '@react-navigation/native';

import { useDispatch } from 'react-redux';
import { addTofavorites, removeFromFavorites } from '../../store/ducks/pokedex';
import { useSelector } from 'react-redux';
const ItemFlatList = (props) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();

  const {
    pokemon: { name, favorite },
  } = props;

  const pokemons = useSelector((state) => state.pokedex.pokemons);

  let isFavorite =
    favorite === undefined
      ? !!pokemons.find((el) => el.name === name)
      : favorite;

  const favoriteIcon = isFavorite ? (
    <Icon size={25} name="star" />
  ) : (
    <Icon size={25} name="star-outline" />
  );

  return (
    <View
      style={{
        justifyContent: 'space-between',
        flexDirection: 'row',
        paddingHorizontal: 15,
        paddingVertical: 10,
        marginHorizontal: 20,
      }}>
      {isFavorite ? (
        <TouchableOpacity onPress={() => dispatch(removeFromFavorites(name))}>
          {favoriteIcon}
        </TouchableOpacity>
      ) : (
        <TouchableOpacity onPress={() => dispatch(addTofavorites(name))}>
          {favoriteIcon}
        </TouchableOpacity>
      )}
      <TouchableOpacity
        onPress={() =>
          navigation.navigate('Details', {
            name,
          })
        }>
        <Text>{name}</Text>
      </TouchableOpacity>
    </View>
  );
};

export { ItemFlatList };
