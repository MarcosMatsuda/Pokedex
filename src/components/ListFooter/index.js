import React from 'react';
import { ActivityIndicator } from 'react-native';
import { Container } from './styles';
const ListFooter = ({ loading }) => {
  if (!loading) {
    return null;
  }
  return (
    <Container>
      <ActivityIndicator animating size="large" />
    </Container>
  );
};

export { ListFooter };
