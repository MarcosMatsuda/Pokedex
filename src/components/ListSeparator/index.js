import React from 'react';
import { View } from 'react-native';

const ListSeparator = () => {
  return (
    <View
      style={{
        height: 1,
        width: '100%',
        backgroundColor: '#CED0CE',
      }}
    />
  );
};

export { ListSeparator };
