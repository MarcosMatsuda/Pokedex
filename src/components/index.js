export { ItemDetails } from './ItemDetails';
export { ItemFlatList } from './ItemFlatList';
export { ListSeparator } from './ListSeparator';
export { ListFooter } from './ListFooter';
