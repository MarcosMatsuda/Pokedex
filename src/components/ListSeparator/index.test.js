import 'react-native';
import React from 'react';
import { ListSeparator } from '~/components';
import { render } from '@testing-library/react-native';

describe('Snapshot testing', () => {
  test('should take Snapshot', () => {
    const tree = render(<ListSeparator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
