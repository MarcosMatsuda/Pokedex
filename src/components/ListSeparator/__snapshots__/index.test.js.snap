// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing should take Snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ced0ce",
        "height": 1,
      },
    ]
  }
/>
`;
